timbuctoo-instancev4/src/test/java/nl/knaw/huygens/timbuctoo/v5/graphql/datafetchers/DefaultSummaryPropDataRetrieverTest.java

```diff
@@ -0,0 +1,138 @@
+    CursorQuad foundQuad1 = quadWithObjectUri("http://example.org/objectFound1");
+    CursorQuad foundQuad2 = quadWithObjectUri("http://example.org/objectFound2");
+    given(quadStore.getQuads("http://example.org/source", "http://example.org/path", Direction.OUT, ""))
+      .willReturn(Lists.newArrayList(foundQuad1, foundQuad2).stream());
+
+    instance.retrieveDefaultProperty(subjectWithUri("http://example.org/source"), quadStore);
+
+    verify(quadStore).getQuads("http://example.org/source", "http://example.org/path", Direction.OUT, "");
+    verify(quadStore).getQuads("http://example.org/objectFound1", "http://example.org/path2", Direction.OUT, "");
+    verifyNoMoreInteractions(quadStore);
+  }
+
+
+  @Test
+  public void retrieveDefaultPropertySearchesUntilItFindsAValue() {
+    List<SummaryProp> defaultProperties = Lists.newArrayList(
+      summaryPropertyWithPath("http://example.org/prop1"),
+      summaryPropertyWithPath("http://example.org/prop2"),
+      summaryPropertyWithPath("http://example.org/prop3")
+    );
+    DefaultSummaryPropDataRetriever instance = new DefaultSummaryPropDataRetriever(defaultProperties);
+    QuadStore quadStore = mock(QuadStore.class);
+    CursorQuad foundQuad = quadWithObjectUri("http://example.org/objectFound1");
+    given(quadStore.getQuads("http://example.org/source", "http://example.org/prop2", Direction.OUT, ""))
+      .willReturn(Lists.newArrayList(foundQuad).stream());
+
+    Optional<CursorQuad> found =
+      instance.retrieveDefaultProperty(subjectWithUri("http://example.org/source"), quadStore);
+
+
+    assertThat(found, is(present()));
+    assertThat(found.get(), is(sameInstance(foundQuad)));
+    verify(quadStore).getQuads("http://example.org/source", "http://example.org/prop1", Direction.OUT, "");
+    verify(quadStore).getQuads("http://example.org/source", "http://example.org/prop2", Direction.OUT, "");
+    verifyNoMoreInteractions(quadStore);
+  }
+
+  @Test
+  public void retrieveDefaultPropertyReturnsAnEmptyOptionWhenNoDefaultPropertiesAreConfigured() {
+    DefaultSummaryPropDataRetriever instance = new DefaultSummaryPropDataRetriever(Lists.newArrayList());
+    QuadStore quadStore = mock(QuadStore.class);
+
+    Optional<CursorQuad> found =
+      instance.retrieveDefaultProperty(subjectWithUri("http://example.org/source"), quadStore);
+
+
+    assertThat(found, not(is(present())));
+    verifyZeroInteractions(quadStore);
+  }
+
+
+
+  private CursorQuad quadWithObjectUri(String objectUri) {
+    CursorQuad foundQuad = mock(CursorQuad.class);
+    given(foundQuad.getObject()).willReturn(objectUri);
+    return foundQuad;
+  }
+
+  private SummaryProp summaryPropertyWithPath(String... path) {
+    return SummaryProp.create(Lists.newArrayList(path), "SimplePath");
+  }
+
+  private SubjectReference subjectWithUri(String uri) {
+    SubjectReference source = mock(SubjectReference.class);
+    given(source.getSubjectUri()).willReturn(uri);
+    return source;
+  }
+
+}
```

Comment:
Add test that verifies if the code also tries the second path that was specified when walking a multi-step path